### PR TITLE
Freebayes regionfix

### PR DESCRIPTION
**Pull request by bwlang**
old method always examined all entries in bam file, 
this method respect the user's entry to target or region fields

### CODE REVIEW

**Comment by bwlang** [comment]
@bgruening  this is that freebayes issue we chatted about earlier today

File: tools/freebayes/freebayes.xml [hunk 1]
```diff
@@ -35,14 +35,25 @@
...
-        grep "^@SQ" |
-        cut -f 2- |
-        awk '{ gsub("^SN:","",$1);
-        gsub("^LN:","",$2);
-        print $1"\t0\t"$2; }' >> regions_all.bed &&
-    #end for
+    ##if user has specified a region or target file, just use instead of calculating a set of unique regions
+
+    #if str( $target_limit_type.target_limit_type_selector ) == "limit_by_target_file":
+        cp '${target_limit_type.input_target_bed}' regions_all.bed &&
```
Comment (bgruening) [inline tools/freebayes/freebayes.xml]: Symlink here?

**Comment by bgruening** [comment]
Thanks @bwlang. LGTM. I just added one small comment.

File: tools/freebayes/freebayes.xml [hunk 2]
```diff
@@ -35,14 +35,25 @@
...
+    #elif str( $target_limit_type.target_limit_type_selector ) == "limit_by_region":
+        echo -e '${target_limit_type.region_chromosome}\t${target_limit_type.region_start}\t${target_limit_type.region_end}' > regions_all.bed &&
```
Comment (yhoogstrate) [inline tools/freebayes/freebayes.xml]: `-e` is not POSIX (http://pubs.opengroup.org/onlinepubs/009695399/utilities/echo.html), how do you thing about using `printf` over `echo -e`?

File: tools/freebayes/freebayes.xml [hunk 3]
```diff
@@ -35,14 +35,25 @@
...
+    #else
+        ##divide up the regions in the bam file for efficient processing
+        #for $bam_count, $input_bam in enumerate( $reference_source.input_bams ):
+            samtools view -H b_${bam_count}.bam |
```
Comment (yhoogstrate) [inline tools/freebayes/freebayes.xml]: Do you think this is simpler and equivalent: `samtools idxstats b_${bam_count}.bam | grep -v '^*' | cut -f 1,2 | sed 's@\t@\t0\t@'` ?

Comment (bwlang) [inline tools/freebayes/freebayes.xml]: certainly simpler code... but it's a more invasive change and I'm not certain we can count on these bams to be indexed 100% of the time.  I'm leaning toward leaving it as is unless there is a strong opinion against that.

**Comment by yhoogstrate** [comment]
Awesome, thanks @bwlang @bgruening and @nsoranzo :+1:

**Pull request merged**